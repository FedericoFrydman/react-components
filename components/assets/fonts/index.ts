import inter_font from './Inter.ttf';
import poppins_Black_font from './Poppins-Black.ttf';
import poppins_BlackItalic_font from './Poppins-BlackItalic.ttf';
import poppins_Bold_font from './Poppins-Bold.ttf';
import poppins_BoldItalic_font from './Poppins-BoldItalic.ttf';
import poppins_ExtraBold_font from './Poppins-ExtraBold.ttf';
import poppins_ExtraBoldItalic_font from './Poppins-ExtraBoldItalic.ttf';
import poppins_ExtraLight_font from './Poppins-ExtraLight.ttf';
import poppins_ExtraLightItalic_font from './Poppins-ExtraLightItalic.ttf';
import poppins_Italic_font from './Poppins-Italic.ttf';
import poppins_Light_font from './Poppins-Light.ttf';
import poppins_LightItalic_font from './Poppins-LightItalic.ttf';
import poppins_Medium_font from './Poppins-Medium.ttf';
import poppins_MediumItalic_font from './Poppins-MediumItalic.ttf';
import poppins_Regular_font from './Poppins-Regular.ttf';
import poppins_SemiBold_font from './Poppins-SemiBold.ttf';
import poppins_SemiBoldItalic_font from './Poppins-SemiBoldItalic.ttf';
import poppins_Thin_font from './Poppins-Thin.ttf';
import poppins_ThinItalic_font from './Poppins-ThinItalic.ttf';

export const FONTS: { [char: string]: string } = {
  inter_font,
  poppins_BlackItalic_font,
  poppins_Black_font,
  poppins_BoldItalic_font,
  poppins_Bold_font,
  poppins_ExtraBoldItalic_font,
  poppins_ExtraBold_font,
  poppins_ExtraLightItalic_font,
  poppins_ExtraLight_font,
  poppins_Italic_font,
  poppins_LightItalic_font,
  poppins_Light_font,
  poppins_MediumItalic_font,
  poppins_Medium_font,
  poppins_Regular_font,
  poppins_SemiBoldItalic_font,
  poppins_SemiBold_font,
  poppins_ThinItalic_font,
  poppins_Thin_font,
};
