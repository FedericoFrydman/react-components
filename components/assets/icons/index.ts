export * from './icons-v1';
export * from './icons-v2';

import { IconsV1, ICON_COLORS_V1, ICON_NAMES_V1 } from './icons-v1';
import { IconsV2, ICON_COLORS_V2, ICON_NAMES_V2 } from './icons-v2';

export type IconSet = { [name: string]: string };

export const getLibraryVerisonByIcon = (name: string): string | null => {
  if (Object.keys(IconsV2).includes(name)) {
    return '2';
  } else if (Object.keys(IconsV1).includes(name)) {
    return '1';
  } else {
    return null;
  }
};

export const ICON_NAMES = [...ICON_NAMES_V1, ...ICON_NAMES_V2];

export const ICON_COLORS_ALL = [...ICON_COLORS_V1, ...ICON_COLORS_V2];

export type IconColors = (typeof ICON_COLORS_ALL)[number];

export const Icons = { ...IconsV1, ...IconsV2 };
