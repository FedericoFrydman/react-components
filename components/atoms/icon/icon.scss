@import '../../assets/styles/brand.scss';
@import '../../assets/styles/colors.scss';

.lsux-icon-container {
  display: flex;
  flex-wrap: wrap;
}

.lsux-icon {
  filter: $filter-default;
  transition: all 0.15s ease-in-out;

  &--no-transition {
    transition: none;
  }

  &.lsux-icon--mono {
    filter: none;
  }

  &--clickable {
    cursor: pointer;
  }

  &--xsmall {
    height: 12px;
    width: 12px;
  }

  &--small {
    height: 16px;
    width: 16px;
  }

  &--medium-small {
    height: 18px;
    width: 18px;
  }

  &--medium {
    height: 24px;
    width: 24px;
  }

  &--large {
    height: 32px;
    width: 32px;
  }

  &--xlarge {
    height: 48px;
    width: 48px;
  }

  &--N00 {
    filter: $filter-white;
  }
  &--N100 {
    filter: $filter-neutral-100;
  }
  &--N300 {
    filter: $filter-neutral-300;
  }
  &--N500 {
    filter: $filter-light-gray;
  }
  &--N700 {
    filter: $filter-gray;
  }
  &--N800 {
    filter: $filter-black;
  }
  &--N1000 {
    filter: $filter-deep-black;
  }

  &--G10 {
    filter: $filter-green-10;
  }
  &--G100 {
    filter: $filter-green-100;
  }
  &--G200 {
    filter: $filter-green;
  }
  &--G300 {
    filter: $filter-green-300;
  }

  &--P10 {
    filter: $filter-purple-10;
  }
  &--P100 {
    filter: $filter-purple-100;
  }
  &--P200 {
    filter: $filter-purple-200;
  }
  &--P300 {
    filter: $filter-purple-300;
  }

  &--R10 {
    filter: $filter-red-10;
  }
  &--R100 {
    filter: $filter-red-100;
  }
  &--R200 {
    filter: $filter-red-200;
  }
  &--R300 {
    filter: $filter-red;
  }

  &--B10 {
    filter: $filter-blue-10;
  }
  &--B100 {
    filter: $filter-blue-100;
  }
  &--B200 {
    filter: $filter-blue;
  }
  &--B300 {
    filter: $filter-blue-300;
  }

  &--Y10 {
    filter: $filter-yellow-10;
  }
  &--Y100 {
    filter: $filter-yellow-100;
  }
  &--Y200 {
    filter: $filter-yellow-200;
  }
  &--Y300 {
    filter: $filter-yellow;
  }

  &--BRAND50 {
    filter: var(--brand-color-prim50-filter);
  }

  &--BRAND100 {
    filter: var(--brand-color-prim100-filter);
  }

  &--BRAND200 {
    filter: var(--brand-color-prim200-filter);
  }

  &--BRAND300 {
    filter: var(--brand-color-prim300-filter);
  }
}

.circle {
  background: $R200;
  border-radius: 50%;

  &--xsmall {
    height: 4px;
    width: 4px;
  }

  &--small {
    height: 5px;
    width: 5px;
  }

  &--medium-small {
    height: 6px;
    width: 6px;
  }

  &--medium {
    height: 7px;
    width: 7px;
  }

  &--large {
    height: 9px;
    width: 9px;
  }

  &--xlarge {
    height: 11px;
    width: 11px;
  }
}
