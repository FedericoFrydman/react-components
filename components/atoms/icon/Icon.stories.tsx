import { Story, Meta } from '@storybook/react';

import { Icon, IconProps } from './Icon';

import docs from './Icon.docs.mdx';
import { ICON_NAMES } from '../../../components/assets/icons';
import { ICON_COLORS_V2 } from '../../../components/assets/icons/icons-v2';
import { Container } from '../container/Container';

export const Default: Story<IconProps> = (args: IconProps) => (
  <Container classNames={['m-3']}>
    <Icon name="action_add" {...args} />
  </Container>
);

export const Clickable = Default.bind({});
Clickable.args = {
  name: 'action_add',
  onClick: () => alert('Add icon clicked'),
};

export const ExtraLargeIconWithBadge = Default.bind({});
ExtraLargeIconWithBadge.args = {
  name: 'action_add',
  size: 'xlarge',
  useBadge: true,
};

export const LargeIconWithBadge = Default.bind({});
LargeIconWithBadge.args = {
  name: 'action_add',
  size: 'large',
  useBadge: true,
};

export const MediumIconWithBadge = Default.bind({});
MediumIconWithBadge.args = {
  name: 'action_add',
  size: 'medium',
  useBadge: true,
};

export const SmallIconWithBadge = Default.bind({});
SmallIconWithBadge.args = {
  name: 'action_add',
  size: 'small',
  useBadge: true,
};

export const XsmallIconWithBadge = Default.bind({});
XsmallIconWithBadge.args = {
  name: 'action_add',
  size: 'xsmall',
  useBadge: true,
};

export default {
  argTypes: {
    color: {
      control: {
        type: 'select',
      },
      options: ICON_COLORS_V2,
    },
    name: { control: { type: 'select' }, options: ICON_NAMES },
    noTransition: { control: { type: 'boolean' } },
    size: { control: { type: 'select' }, options: ['xsmall', 'small', 'medium-small', 'medium', 'large', 'xlarge'] },
    useBadge: { control: { type: 'boolean' } },
  },
  component: Default,
  parameters: {
    docs: { page: docs },
  },
  title: 'Components/Icon',
} as Meta;
