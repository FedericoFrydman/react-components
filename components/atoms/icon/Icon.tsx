import { forwardRef, Ref } from 'react';
import cn from 'classnames';

import './icon.scss';
import { Container } from '../container/Container';
import { Icons } from '../../assets/icons';
import { IconColorsV1, IconNamesV1 } from '../../assets/icons/icons-v1';
import { IconColorsV2, IconNamesV2 } from '../../assets/icons/icons-v2';

export type IconLibraryVersion = '1' | '2';

export type IconNames = IconNamesV1 | IconNamesV2;

export type IconAndBadgeSizes = 'xsmall' | 'small' | 'medium-small' | 'medium' | 'large' | 'xlarge';

export const excludedCategories: string[] = ['apps', 'badges', 'logo', 'logos', 'feedback', 'payments', 'medialogo'];

export interface IconProps
  extends Omit<React.HTMLProps<HTMLImageElement>, 'css' | 'alt' | 'src' | 'size' | 'crossOrigin'> {
  /**
   * Applies a color filter to the icon.
   */
  color?: IconColorsV1 | IconColorsV2;
  /**
   * Name of the icon to be shown.
   */
  name: IconNamesV1 | IconNamesV2;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * Defines icon size and badge size.
   */
  size?: IconAndBadgeSizes;
  /**
   * Disables CSS transition on the icon.
   */
  noTransition?: boolean;
  /**
   * Additional data that is dispatched with the tracking event.
   */
  onClick?: (e: React.MouseEvent) => void | unknown;
  /**
   * Alt tag for the HTML element
   * @default `name` paramater
   */
  alt?: string;
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLImageElement>;
  /**
   * Define if icon has a Badge.
   */
  useBadge?: boolean;
}

const IconComponent = (
  {
    classNames = [],
    color = 'default',
    name,
    alt = name + '.',
    onClick,
    size = 'medium',
    noTransition,
    useBadge = false,
    ...props
  }: IconProps,
  ref: IconProps['ref'],
) => {
  const classIcon = 'lsux-icon';
  const classIconContainer = 'lsux-icon-container';

  const applyMonoClass = () => {
    let isMono = false;
    excludedCategories.forEach((c) => {
      if (name?.includes(`${c}_`)) isMono = true;
    });
    return isMono ? `${classIcon}--mono` : '';
  };

  const cnIconContainer = [`${classIconContainer}`, `flex-row`];

  const cnIcon = cn(
    {
      [`${classIcon}`]: true,
      [`${classIcon}--${color}`]: color !== 'default',
      [`${classIcon}--${size}`]: size,
      [`${classIcon}--no-transition`]: noTransition,
      [`${classIcon}--clickable`]: !!onClick,
      [applyMonoClass()]: true,
    },
    classNames,
  );

  const cnBadge = cn({
    [`circle`]: true,
    [`circle--${size}`]: size,
  });

  if (name === null || name === undefined) {
    return <div onClick={onClick} className={cnIcon}></div>;
  } else {
    return (
      <Container classNames={cnIconContainer} background="none" flexbox flexDirection="row">
        <img {...props} src={Icons[name] as string} alt={alt} className={cnIcon} ref={ref} onClick={onClick}></img>
        {useBadge && <div className={cnBadge} />}
      </Container>
    );
  }
};

export const Icon = forwardRef<HTMLImageElement, IconProps>(IconComponent);
