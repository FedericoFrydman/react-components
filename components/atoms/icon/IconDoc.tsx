import { FC } from 'react';
import { ICON_NAMES_V1 } from '../../assets/icons/icons-v1';
import { ICON_NAMES_V2 } from '../../assets/icons/icons-v2';
import { Heading } from '../heading/Heading';
import { Icon, IconLibraryVersion } from './Icon';
import { Text } from '../text/Text';

const groupByCategories = (libraryVersion: IconLibraryVersion): Record<string, string[]>[] => {
  const category: Record<string, string[]> = {};
  const result: Record<string, string[]>[] = [];

  const ICON_NAMES = libraryVersion === '1' ? ICON_NAMES_V1 : ICON_NAMES_V2;

  ICON_NAMES.forEach((icon) => {
    let key: string = icon.split('_')[0];

    if (libraryVersion === '1') {
      switch (key) {
        case 'action':
          key = 'Action';
          break;
        case 'alert':
          key = 'Alert';
          break;
        case 'comm':
          key = 'Communication';
          break;
        case 'cs':
          key = 'Commercial Shield';
          break;
        case 'file':
          key = 'File';
          break;
        case 'id':
          key = 'IDShield';
          break;
        case 'logo':
          key = 'Logos';
          break;
        case 'ls':
          key = 'Legalshield';
          break;
        case 'nav':
          key = 'Navigation';
          break;
        case 'object':
          key = 'Object';
          break;
        case 'payment':
          key = 'Payment';
          break;
        default:
          break;
      }
    } else if (libraryVersion === '2') {
      switch (key) {
        case 'apps':
          key = 'Apps';
          break;
        case 'arrows':
          key = 'Arrows';
          break;
        case 'badges':
          key = 'Badges';
          break;
        case 'calendar':
          key = 'Calendar';
          break;
        case 'comm':
          key = 'Communication';
          break;
        case 'edit':
          key = 'Edit';
          break;
        case 'feedback':
          key = 'Feedback';
          break;
        case 'file':
          key = 'File';
          break;
        case 'filetypes':
          key = 'File Types';
          break;
        case 'interface':
          key = 'Interface';
          break;
        case 'live':
          key = 'Live';
          break;
        case 'logos':
          key = 'Logos';
          break;
        case 'marks':
          key = 'Marks';
          break;
        case 'media':
          key = 'Media';
          break;
        case 'medialogo':
          key = 'Media Logos';
          break;
        case 'menu':
          key = 'Menu';
          break;
        case 'nav':
          key = 'Navigation';
          break;
        case 'partnerships':
          key = 'Partnerships';
          break;
        case 'payments':
          key = 'Payments';
          break;
        case 'social':
          key = 'Social';
          break;
        case 'system':
          key = 'System';
          break;
        case 'user':
          key = 'User';
          break;
        case 'warning':
          key = 'Warning';
          break;
        default:
          break;
      }
    }

    if (!category[key]) {
      category[key] = [];
      result.push({ [key]: category[key] });
    }
    category[key].push(icon);
  });
  return result;
};

export const IconDoc: FC<{ libraryVersion: IconLibraryVersion }> = ({
  libraryVersion,
}: {
  libraryVersion: IconLibraryVersion;
}) => {
  return (
    <>
      {groupByCategories(libraryVersion).map((category) => {
        const title = Object.keys(category)[0];
        return (
          <div key={title} className="mb-3">
            <Heading as="T16" text={title} />
            <div style={{ display: 'flex', flexWrap: 'wrap' }}>
              {category[title].map((icon) => {
                return (
                  <div key={icon} className="m-5" style={{ textAlign: 'center', width: '7.5rem' }}>
                    <Icon id={icon} size="large" name={icon} />
                    <label htmlFor={icon} style={{ display: 'block' }}>
                      <Text textSize="small" text={icon} />
                    </label>
                  </div>
                );
              })}
            </div>
          </div>
        );
      })}
    </>
  );
};
