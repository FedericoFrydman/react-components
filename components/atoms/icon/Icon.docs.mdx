import { Meta, ArgsTable, Story, Canvas, Title, Description } from '@storybook/addon-docs';

import { IconDoc } from './IconDoc';
import { Icon } from './Icon';
import { Heading } from '../heading/Heading';

<Title />

<Description />

<Canvas>
  <Story id="components-icon--default" />
</Canvas>

<Meta />

<ArgsTable of={Icon} />

## Sizes

| <Heading as="T16" text="XSmall" />          | <Heading as="T16" text="Small" />          | <Heading as="T16" text="Medium Small" />          | <Heading as="T16" text="Medium" />          | <Heading as="T16" text="Large" />          | <Heading as="T16" text="XLarge" />          |
| ------------------------------------------- | ------------------------------------------ | ------------------------------------------------- | ------------------------------------------- | ------------------------------------------ | ------------------------------------------- |
| <Heading as="T16" text="12x12 px" />        | <Heading as="T16" text="16x16 px" />       | <Heading as="T16" text="18x18 px" />              | <Heading as="T16" text="24x24 px" />        | <Heading as="T16" text="32x32 px" />       | <Heading as="T16" text="48x48 px" />        |
| <Icon name="system_camera" size="xsmall" /> | <Icon name="system_camera" size="small" /> | <Icon name="system_camera" size="medium-small" /> | <Icon name="system_camera" size="medium" /> | <Icon name="system_camera" size="large" /> | <Icon name="system_camera" size="xlarge" /> |

## Colors

After version v1.11.0 icons have the option to use any color in our [color palette](?path=/docs/foundation-colors--page)

**For old color references:**

| Old color name | New color name |
| -------------- | -------------- |
| blue           | B200           |
| green          | G200           |
| light-gray     | N500           |
| gray           | N700           |
| black          | N800           |
| deep-black     | N1000          |
| red            | R300           |
| yellow         | Y300           |

These is the list for all colors now available for ICON_COLORS

| Color name | Palette color reference | HEX code |
| ---------- | ----------------------- | -------- |
| default    |
| B10        | $B10                    | #e9eff9  |
| B100       | $B100                   | #377bdd  |
| B200       | $B200                   | #1952bd  |
| B200       | $B200                   | #13429a  |
| G10        | $G10                    | #e9f4ed  |
| G100       | $G100                   | #34ab5e  |
| G200       | $G200                   | #2c9150  |
| G300       | $G300                   | #006e3c  |
| N00        | $N00                    | #ffffff  |
| N100       | $N100                   | #f7f7f7  |
| N300       | $N300                   | #eeeeee  |
| N500       | $N500                   | #afafaf  |
| N700       | $N700                   | #545454  |
| N800       | $N800                   | #222222  |
| N1000      | $N1000                  | #000000  |
| P10        | $P10                    | #f0ecf3  |
| P100       | $P100                   | #a073bc  |
| P200       | $P200                   | #70468c  |
| P300       | $P300                   | #431c5e  |
| R10        | $R10                    | #ffeef0  |
| R100       | $R100                   | #ce0220  |
| R200       | $R200                   | #b5021c  |
| R300       | $R300                   | #920217  |
| Y10        | $Y10                    | #fffee6  |
| Y100       | $Y100                   | #ffd633  |
| Y200       | $Y200                   | #ffcc00  |
| Y300       | $Y300                   | #e6b800  |
| BRAND50    | name                    | NA       |
| BRAND100   | name                    | NA       |
| BRAND200   | name                    | NA       |
| BRAND300   | name                    | NA       |

## Events

An icon has all the properties of an img tag, so it can receive events. Particularly the **onClick** event adds the `cursor: pointer` style.

<Canvas>
  <Story id="components-icon--clickable" />
</Canvas>

## Icons

Below are icons use included for responsive web available in different icon colors and icon sizes. For iOS and Android app, "System" icons are the set from the OS itself. We took advantage to use them for a cohesive app experience and easy to maintenance.

## Version 2 Icons

<IconDoc libraryVersion="2" />

## Version 1 Icons

<IconDoc libraryVersion="1" />
