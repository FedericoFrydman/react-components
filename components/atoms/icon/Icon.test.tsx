import { createRef } from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { render, screen } from '@testing-library/react';
import cn from 'classnames';

import { Icon, IconAndBadgeSizes, IconProps } from './Icon';
import { ICON_NAMES } from '../../../components/assets/icons';
import { excludedCategories } from '../../atoms/icon/Icon';

describe('Icon Component', () => {
  let baseClass: string;
  let testId: string;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    baseClass = 'lsux-icon';
    testId = 'test icon';
    wrapper = shallow(<Icon name="action_add" />);
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('props', () => {
    describe('name', () => {
      describe('if name is in excluded category', () => {
        const monoIconNames = ICON_NAMES.filter((n) => excludedCategories.includes(n.split('_')[0]));

        function expectMonoClass(iconName: string) {
          it('does contain mono class', () => {
            const icon = renderIcon({ name: iconName });
            expect(icon).toHaveClass(`${baseClass}--mono`, { exact: false });
          });
        }

        monoIconNames.forEach((name) => {
          expectMonoClass(name);
        });
      });

      describe('if name is not in excluded category', () => {
        const nonMonoIconNames = ICON_NAMES.filter((n) => !excludedCategories.includes(n.split('_')[0]));

        function expectNonMonoClass(iconName: string) {
          it('does not contain mono class', () => {
            const icon = renderIcon({ name: iconName });
            expect(icon).not.toHaveClass(`${baseClass}--mono`, { exact: false });
          });
        }

        nonMonoIconNames.forEach((name) => {
          expectNonMonoClass(name);
        });
      });
    });

    describe('noTransition', () => {
      describe('when noTransition is true', () => {
        it('does contain no transition class', () => {
          const icon = renderIcon({ name: 'action_add', noTransition: true });
          expect(icon).toHaveClass(`${baseClass}--no-transition`, { exact: false });
        });
      });

      describe('when noTransition is not present', () => {
        it('does not contain the no transition class', () => {
          const icon = renderIcon({ name: 'action_add' });
          expect(icon).not.toHaveClass(`${baseClass}--no-transition`, { exact: false });
        });
      });
    });

    describe('size', () => {
      const iconSizes: IconAndBadgeSizes[] = ['xsmall', 'small', 'medium-small', 'medium', 'large', 'xlarge'];
      const iconName = 'calendar';
      let iconSizeClass: string;

      describe('when size is present', () => {
        it('defaults "size" to the "medium"', () => {
          const iconEl = renderIcon({ name: iconName });
          expect(iconEl).toHaveClass(`${baseClass}--medium`);
        });

        function itAssignsTheRightClassForIconSize(iconSize: IconAndBadgeSizes) {
          describe(`when "size" is "${iconSize}"`, () => {
            beforeEach(() => {
              iconSizeClass = `${baseClass}--${iconSize}`;
            });

            it(`assigns the label the ${iconSizeClass} class`, () => {
              const iconEl = renderIcon({ name: iconName, size: iconSize });
              expect(iconEl).toHaveClass(iconSizeClass);
            });
          });
        }

        iconSizes.forEach((iconSize) => itAssignsTheRightClassForIconSize(iconSize));
      });
    });

    describe('classNames', () => {
      describe('when provided classNames', () => {
        const classNames = ['my-class-1', 'my-class-2'];

        it('gives the icon the provided classNames', () => {
          const icon = renderIcon({ classNames, name: 'action_add' });

          expect(icon).toHaveClass(cn(classNames), { exact: false });
        });
      });
    });

    describe('onClick', () => {
      describe('when onClick is present', () => {
        const onClickSpy = jest.fn();

        beforeEach(() => jest.clearAllMocks());

        it('adds the clickable class to the icon', () => {
          const icon = renderIcon({ name: 'action_add', onClick: onClickSpy });
          const onClickClass = `${baseClass}--clickable`;

          expect(icon).toHaveClass(onClickClass, { exact: false });
        });

        it('calls the onClick handler when pressed', () => {
          const icon = renderIcon({ name: 'calendar', onClick: onClickSpy });

          icon.click();

          expect(onClickSpy).toHaveBeenCalledTimes(1);
        });
      });
    });
  });

  describe('ref', () => {
    const ref = createRef<HTMLImageElement>();

    it('provides a ref to the text', () => {
      expect(renderIcon({ name: 'calendar', ref })).toEqual(ref.current);
    });
  });

  describe('unspecified html attributes', () => {
    const altText = 'My accessible button';

    it('allows html attributes to be passed to the underlying button element', () => {
      renderIcon({ alt: altText, name: 'calendar' } as IconProps);

      const icon = screen.getByTestId(testId);

      expect(icon).toHaveAttribute('alt', altText);
    });
  });

  const renderIcon = (props: IconProps): HTMLElement => {
    render(<Icon {...props} data-testid={testId} />);

    return screen.getByTestId(testId);
  };
});
