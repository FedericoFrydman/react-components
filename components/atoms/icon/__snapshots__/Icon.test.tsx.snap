// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should render the component 1`] = `
<ForwardRef(ContainerComponent)
  background="none"
  classNames={
    Array [
      "lsux-icon-container",
      "flex-row",
    ]
  }
  flexDirection="row"
  flexbox={true}
>
  <img
    alt="action_add."
    className="lsux-icon lsux-icon--medium "
    src={[Function]}
  />
</ForwardRef(ContainerComponent)>
`;
